pkg/agent/config.go

```diff
@@ -66,6 +66,9 @@ type Config struct {
 
 	// Telemetry provides the configuration for metrics exporting
 	Telemetry telemetry.FileConfig
+
+	// UptimeReportInterval the interval that spire agent reports its uptime.
+	UptimeReportInterval time.Duration
```

Comment:
Do you think that it will be commonly needed to override the report interval or this is added just in case is needed? If there is not a specific use case at this point, I would be rather inclined to not have it. New settings means new possible misconfigurations, documentation changes and possible deprecation handling in the future. We can always add it in the future if we find a reason, but it's more difficult to remove it.

If we do think it's needed, then other changes in this PR are needed:
1) This setting needs to be parsed from the agent's config file in https://github.com/spiffe/spire/blob/master/cmd/spire-agent/cli/run/run.go, otherwise it's an unknown setting for the agent.
2) https://github.com/spiffe/spire/blob/master/doc/spire_agent.md needs to be updated to document this setting.
3) https://github.com/spiffe/spire/blob/master/conf/agent/agent_full.conf needs to be updated with this setting.

Reply:
my personal opinion would be this should be `just in case` needed. 

@amoore877 any comment on this?

Reply:
Given that spire agents are generally widely deployed, at scale this amounts to a lot of metrics data. In the worst case, it would be nice to have this be configurable in case of needs to fine-tune (or turn off) to not overload an observability system in case the sane default is too aggressive.
Other cases may be different operators have different needs and standards for granularity of the metric.

For misconfiguration risk, we could add a sane minimum (10s, 20s?).

Would it make sense to instead make this an "experimental" configurable? There have been such experimental (or hidden) config in the past for features.

Reply:
It may also make sense for server and agent to be configured for different intervals.

Reply:
We discussed this in today's maintainer's call and the consensus was that adding a new configurable is not needed. The introduction of this metric doesn't seem to really affect the total amount of metrics emitted and adding config settings for this looks very targeted for this case considering all the metrics that we already have.
We also discussed about the reporting interval and we feel that a 10 seconds interval seems a reasonable value to start with.

Reply:
@amartinezfayo  thanks for your detailed info. I updated the code and removed the configurable value.